ethcore/evm/src/instructions.rs

```diff
@@ -586,6 +592,9 @@ lazy_static! {
 		arr[LOG2 as usize] = Some(InstructionInfo::new("LOG2", 4, 0, GasPriceTier::Special));
 		arr[LOG3 as usize] = Some(InstructionInfo::new("LOG3", 5, 0, GasPriceTier::Special));
 		arr[LOG4 as usize] = Some(InstructionInfo::new("LOG4", 6, 0, GasPriceTier::Special));
+		arr[BEGINSUB as usize] = Some(InstructionInfo::new("BEGINSUB", 0, 0, GasPriceTier::Base));
+		arr[JUMPSUB as usize] = Some(InstructionInfo::new("JUMPSUB", 1, 0, GasPriceTier::Low));
```

Comment:
Unless I'm mistaken, both geth and besu used `mid` (8) for `JUMPSUB`.

Reply:
We followed the suggestion from the spec
> We suggest that the cost of JUMPSUB be low, and RETURNSUB be verylow. Measurement will tell.

Is there any reason why geth and besu chose `mid`?

Reply:
I think the original geth-PR from Greg used that value, and then we didn't change it when the EIP was updated at some point. IMO it makes zero sense to have `JUMPSUB` be cheaper than `JUMP`, considering that `JUMPSUB` causes a memory allocation in the returnstack.

Reply:
PTAL: https://github.com/ethereum/EIPs/pull/2669